src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.h

```diff
@@ -16,40 +16,97 @@
 #ifndef _SW_EMU_SHIM_H_
 #define _SW_EMU_SHIM_H_
 
-#include "unix_socket.h"
 #include "config.h"
+#include "core/common/api/xclbin_int.h"
+#include "core/common/device.h"
+#include "core/common/message.h"
+#include "core/common/scheduler.h"
+#include "core/common/query_requests.h"
+#include "core/common/xrt_profiling.h"
+#include "core/include/experimental/xrt_xclbin.h"
 #include "em_defines.h"
 #include "memorymanager.h"
 #include "rpc_messages.pb.h"
-
+#include "swscheduler.h"
+#include "unix_socket.h"
+#include "xclbin.h"
 #include "xclperf.h"
 #include "xcl_api_macros.h"
 #include "xcl_macros.h"
-#include "xclbin.h"
-#include "core/common/device.h"
-#include "core/common/scheduler.h"
-#include "core/common/message.h"
-#include "core/common/xrt_profiling.h"
-#include "core/common/query_requests.h"
-#include "core/common/api/xclbin_int.h"
-#include "core/include/experimental/xrt_xclbin.h"
 
-#include "swscheduler.h"
+#include <fcntl.h>
 #include <stdarg.h>
 #include <sys/mman.h>
-#include <sys/types.h>
 #include <sys/stat.h>
-#include <fcntl.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+
+#include<atomic>
 #include <thread>
 #include <tuple>
-#include <sys/wait.h>
+
 #ifndef _WINDOWS
 #include <dlfcn.h>
 #endif
 
 namespace xclcpuemhal2 {
   using key_type = xrt_core::query::key_type;
   const uint64_t MEMSIZE = 0x0000000080000000;
+  /**
+   * sParseLog structure parses a file named mFileName and looks for a matchString
+   * On successfull match, print the line to the console
+   * Currently, we are using this structure to parse the external IO file that is
+   * generated by the deviceProcess during SW EMU.
+   */
+  struct sParseLog
+  {
+    std::ifstream file;
+    std::string mFileName;
+    std::atomic_bool mFileExists;
+
+    sParseLog(const std::string& iDeviceLog):mFileName(iDeviceLog) {
+      mFileExists = false;    
+    }  
+
+    ~sParseLog() = default;
+    //**********************************************************************************//
+    /**
+    * printMsg(std::string&) - Searches for a matchString in a file .
+    * On a successfull match, it prints an actionable suggestion on the console.
+    *
+    * @matchString:    string to match
+    * 
+    */
+    void printMsg(std::string &matchString) {
+      std::string line;
+      while (std::getline(file, line)) {
+        if (line.find(matchString) != std::string::npos)
+          std::cout << "Received request to end the application. Press Cntrl+C to exit the application." << std::endl; 
+      }
+    }
+
+    //**********************************************************************************//
+    /**
+    * parseLog() - Checks for file existence and calls printMsg .
+    * 
+    */
+    void parseLog() {
+      if (!mFileExists) {
```

Comment:
`mFileExists` has nothing to do with the existence of the file if I understand correctly.